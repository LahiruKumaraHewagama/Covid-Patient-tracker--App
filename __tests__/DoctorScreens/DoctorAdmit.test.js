import React from "react";
import DoctorAdmit from "../../screens/DoctorScreens/DoctorAdmit";
import { act, create } from "react-test-renderer";
import { render, fireEvent, waitFor, cleanup } from '@testing-library/react-native';

jest.useFakeTimers();
const navigation = { addListener: jest.fn() }
afterEach(cleanup);
const isPickerShow = true;

let tree
test('snapshot Doctor Admit page', async () => {  
    act(() => { tree = create(<DoctorAdmit navigation={navigation} {...isPickerShow}/>) })   
        expect(tree).toMatchSnapshot();
    })


test('Name empty input error massage', async () => {
    global.alert = jest.fn();
    const admitBtn = tree.root.findByProps({ testID: 'admit' }).props;
    act(() => admitBtn.onPress());
    expect(global.alert).toHaveBeenCalledWith("Name can't be empty !");
});

test('Birth Day empty input error massage', async () => {
    global.alert = jest.fn();
    const name = tree.root.findByProps({ testID: 'name_test' }).props;
    act(() => name.onChangeText("Lahiu Nuraj"));

    const admitBtn = tree.root.findByProps({ testID: 'admit' }).props;
    act(() => admitBtn.onPress());

    expect(global.alert).toHaveBeenCalledWith("Date of Birthday can't be empty !");

});


// test('Contactnumber empty input error massage', async () => {
//     global.alert = jest.fn();
//     const name = tree.root.findByProps({ testID: 'name_test' }).props;
//     act(() => name.onChangeText("Lahiu Nuraj"));

//     const bday = tree.root.findByProps({ testID: 'bday_test' }).props;

//     act(() => bday.onChange(new Date() ));

//     const admitBtn = tree.root.findByProps({ testID: 'admit' }).props;
//     act(() => admitBtn.onPress());

//     expect(global.alert).toHaveBeenCalledWith("Contactnumber can't be empty !");

// });

// test('District empty input error massage', async () => {
//     global.alert = jest.fn();
//     const name = tree.root.findByProps({ testID: 'name_test' }).props;
//     act(() => name.onChangeText("Lahiu Nuraj"));

//     const bday = tree.root.findByProps({ testID: 'bday_test' }).props;
//     act(() => bday.onChangeText("2011-01-11"));

//     const phone = tree.root.findByProps({ testID: 'phone_test' }).props;
//     act(() => phone.onChangeText("0710158123"));

//     const admitBtn = tree.root.findByProps({ testID: 'admit' }).props;
//     act(() => admitBtn.onPress());

//     expect(global.alert).toHaveBeenCalledWith("District can't be empty !");

// });


// test('District empty input error massage', async () => {
//     global.alert = jest.fn();
//     const name = tree.root.findByProps({ testID: 'name_test' }).props;
//     act(() => name.onChangeText("Lahiu Nuraj"));

//     const bday = tree.root.findByProps({ testID: 'bday_test' }).props;
//     act(() => bday.onChangeText("2011-01-11"));

//     const phone = tree.root.findByProps({ testID: 'phone_test' }).props;
//     act(() => phone.onChangeText("0710158123"));

//     const district = tree.root.findByProps({ testID: 'district_test' }).props;
//     act(() => district.onChangeText("Matara"));

//     const admitBtn = tree.root.findByProps({ testID: 'admit' }).props;
//     act(() => admitBtn.onPress());

//     expect(global.alert).toHaveBeenCalledWith("Please select RATresult !");

// });




