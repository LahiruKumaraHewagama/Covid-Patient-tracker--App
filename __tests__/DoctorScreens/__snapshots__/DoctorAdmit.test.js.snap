// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`snapshot Doctor Admit page 1`] = `
<RCTSafeAreaView
  emulateUnlessSupported={true}
  style={
    Object {
      "backgroundColor": "#fff",
      "flex": 3,
      "paddingHorizontal": 20,
      "paddingVertical": 20,
    }
  }
>
  <View
    style={
      Object {
        "justifyContent": "flex-end",
        "paddingBottom": 20,
        "paddingHorizontal": 20,
      }
    }
  >
    <Text
      style={
        Object {
          "color": "#009387",
          "fontSize": 30,
          "fontWeight": "bold",
          "textAlign": "center",
        }
      }
    >
      ADMIT NEW PATIENT 
    </Text>
    <View
      style={
        Object {
          "borderBottomColor": "#009387",
          "borderBottomWidth": 2,
        }
      }
    />
  </View>
  <RCTScrollView
    style={
      Object {
        "paddingRight": 20,
      }
    }
  >
    <View>
      <Text
        style={
          Object {
            "color": "#007c7a",
            "fontSize": 16,
          }
        }
      >
        Patient Name
      </Text>
      <View
        style={
          Object {
            "alignItems": "center",
            "borderColor": "#007c7a",
            "borderRadius": 20,
            "borderWidth": 1,
            "flexDirection": "row",
            "marginTop": 5,
            "padding": 5,
          }
        }
      >
        <TextInput
          allowFontScaling={true}
          autoCapitalize="none"
          onChangeText={[Function]}
          placeholder="Enter name"
          placeholderTextColor="#666666"
          rejectResponderTermination={true}
          style={
            Object {
              "color": "#000",
              "flex": 1,
              "fontSize": 15,
              "paddingLeft": 20,
            }
          }
          testID="name_test"
          underlineColorAndroid="transparent"
          value=""
        />
      </View>
      <View
        style={
          Object {
            "flexDirection": "row",
            "marginTop": 15,
          }
        }
      >
        <Text
          style={
            Object {
              "color": "#007c7a",
              "fontSize": 16,
            }
          }
        >
          Date of Birth
        </Text>
      </View>
      <View
        onPress={[Function]}
        style={
          Object {
            "alignItems": "center",
            "borderColor": "#007c7a",
            "borderRadius": 20,
            "borderWidth": 1,
            "flex": 0.5,
            "flexDirection": "row",
            "marginTop": 10,
            "padding": 5,
            "paddingLeft": 10,
          }
        }
      >
        <Text />
        <Text
          style={
            Object {
              "color": "black",
              "fontSize": 15,
              "paddingLeft": 5,
            }
          }
        >
          
        </Text>
        <TextInput
          allowFontScaling={true}
          autoCapitalize="none"
          editable={false}
          placeholder="Select Date"
          placeholderTextColor="#666666"
          rejectResponderTermination={true}
          style={
            Object {
              "color": "#000",
              "flex": 1,
              "fontSize": 15,
              "paddingLeft": 20,
            }
          }
          underlineColorAndroid="transparent"
        />
      </View>
      <Text
        style={
          Array [
            Object {
              "marginTop": 15,
            },
            Object {
              "color": "#007c7a",
              "fontSize": 16,
            },
          ]
        }
      >
        Contact Number
      </Text>
      <View
        style={
          Object {
            "alignItems": "center",
            "borderColor": "#007c7a",
            "borderRadius": 15,
            "borderWidth": 1,
            "flexDirection": "row",
            "marginTop": 10,
            "padding": 5,
          }
        }
      >
        <View
          style={
            Array [
              Object {
                "backgroundColor": "white",
                "flexDirection": "row",
                "width": 600,
              },
              Object {},
              Object {
                "alignSelf": "center",
                "height": 30,
                "marginTop": 1,
                "width": 133,
              },
            ]
          }
        >
          <View
            accessible={true}
            focusable={true}
            onClick={[Function]}
            onResponderGrant={[Function]}
            onResponderMove={[Function]}
            onResponderRelease={[Function]}
            onResponderTerminate={[Function]}
            onResponderTerminationRequest={[Function]}
            onStartShouldSetResponder={[Function]}
            style={
              Object {
                "alignItems": "center",
                "flexDirection": "row",
                "height": "100%",
                "justifyContent": "center",
                "minWidth": 32,
                "opacity": 1,
                "width": 150,
              }
            }
          >
            <View
              style={
                Object {
                  "alignItems": "center",
                  "justifyContent": "center",
                  "marginRight": 10,
                  "width": 30,
                }
              }
            >
              <ActivityIndicator
                animating={true}
                color="#999999"
                hidesWhenStopped={true}
                size="small"
              />
            </View>
            <Modal
              animated={true}
              animationType="slide"
              hardwareAccelerated={false}
              onDismiss={[Function]}
              onRequestClose={[Function]}
              visible={false}
            >
              <RCTSafeAreaView
                emulateUnlessSupported={true}
                style={
                  Array [
                    Object {
                      "flex": 1,
                    },
                    Object {
                      "backgroundColor": "#ffffff",
                    },
                  ]
                }
              >
                <View
                  style={
                    Object {
                      "alignItems": "center",
                      "flexDirection": "row",
                    }
                  }
                >
                  <View
                    style={
                      Array [
                        Object {
                          "alignItems": "center",
                          "height": 48,
                          "justifyContent": "center",
                          "width": "15%",
                        },
                        undefined,
                      ]
                    }
                  >
                    <View
                      accessible={true}
                      focusable={true}
                      onClick={[Function]}
                      onResponderGrant={[Function]}
                      onResponderMove={[Function]}
                      onResponderRelease={[Function]}
                      onResponderTerminate={[Function]}
                      onResponderTerminationRequest={[Function]}
                      onStartShouldSetResponder={[Function]}
                      style={
                        Object {
                          "opacity": 1,
                        }
                      }
                    >
                      <Image
                        source={1}
                        style={
                          Array [
                            Object {
                              "height": 25,
                              "resizeMode": "contain",
                              "width": 25,
                            },
                            undefined,
                            Object {
                              "tintColor": "#000000",
                            },
                          ]
                        }
                      />
                    </View>
                  </View>
                  <TextInput
                    allowFontScaling={true}
                    autoCorrect={false}
                    autoFocus={false}
                    onBlur={[Function]}
                    onChangeText={[Function]}
                    onClose={[Function]}
                    onFocus={[Function]}
                    placeholder="Enter country name"
                    placeholderTextColor="#aaa"
                    rejectResponderTermination={true}
                    renderFilter={[Function]}
                    style={
                      Array [
                        Object {
                          "height": 48,
                          "width": "70%",
                        },
                        Object {
                          "color": "#000000",
                          "fontFamily": "System",
                          "fontSize": 16,
                        },
                      ]
                    }
                    testID="text-input-country-filter"
                    underlineColorAndroid="transparent"
                    value=""
                    withCloseButton={true}
                    withFilter={true}
                  />
                </View>
                <View
                  style={
                    Array [
                      Object {
                        "flex": 1,
                        "flexDirection": "row",
                        "justifyContent": "space-between",
                      },
                      Object {
                        "backgroundColor": "#ffffff",
                      },
                    ]
                  }
                >
                  <RCTScrollView
                    ItemSeparatorComponent={[Function]}
                    automaticallyAdjustContentInsets={false}
                    data={Array []}
                    disableVirtualization={false}
                    getItem={[Function]}
                    getItemCount={[Function]}
                    getItemLayout={[Function]}
                    horizontal={false}
                    initialNumToRender={14}
                    keyExtractor={[Function]}
                    keyboardShouldPersistTaps="handled"
                    maxToRenderPerBatch={10}
                    onContentSizeChange={[Function]}
                    onEndReachedThreshold={2}
                    onLayout={[Function]}
                    onMomentumScrollEnd={[Function]}
                    onScroll={[Function]}
                    onScrollBeginDrag={[Function]}
                    onScrollEndDrag={[Function]}
                    onScrollToIndexFailed={[Function]}
                    removeClippedSubviews={false}
                    renderItem={[Function]}
                    scrollEventThrottle={1}
                    stickyHeaderIndices={Array []}
                    testID="list-countries"
                    updateCellsBatchingPeriod={50}
                    viewabilityConfigCallbackPairs={Array []}
                    windowSize={21}
                  >
                    <View />
                  </RCTScrollView>
                </View>
              </RCTSafeAreaView>
            </Modal>
            <Image
              resizeMode="contain"
              source={
                Object {
                  "uri": "data:image/png;base64,iVBORw0KGgoAAAANSUhEUgAAACAAAAAgCAYAAABzenr0AAAAi0lEQVRYR+3WuQ6AIBRE0eHL1T83FBqU5S1szdiY2NyTKcCAzU/Y3AcBXIALcIF0gRPAsehgugDEXnYQrUC88RIgfpuJ+MRrgFmILN4CjEYU4xJgFKIa1wB6Ec24FuBFiHELwIpQxa0ALUId9wAkhCnuBdQQ5ngP4I9wxXsBDyJ9m+8y/g9wAS7ABW4giBshQZji3AAAAABJRU5ErkJggg==",
                }
              }
              style={
                Object {
                  "height": 14,
                  "width": 12,
                }
              }
            />
          </View>
          <View
            style={
              Array [
                Object {
                  "alignItems": "center",
                  "backgroundColor": "#F8F9F9",
                  "flex": 1,
                  "flexDirection": "row",
                  "paddingHorizontal": 30,
                  "paddingVertical": 27,
                  "textAlign": "left",
                },
                Object {},
              ]
            }
          >
            <TextInput
              allowFontScaling={true}
              editable={true}
              keyboardAppearance="default"
              keyboardType="number-pad"
              onChangeText={[Function]}
              placeholder="Phone Number"
              rejectResponderTermination={true}
              selectionColor="black"
              style={
                Array [
                  Object {
                    "color": "#000000",
                    "flex": 1,
                    "fontSize": 16,
                  },
                  Object {},
                ]
              }
              underlineColorAndroid="transparent"
              value=""
            />
          </View>
        </View>
        <TextInput
          allowFontScaling={true}
          autoCorrect={false}
          keyboardType="numeric"
          onChangeText={[Function]}
          placeholder="Enter Contact Number "
          placeholderTextColor="#666666"
          rejectResponderTermination={true}
          style={
            Object {
              "color": "#000",
              "flex": 1,
              "fontSize": 15,
              "marginTop": 1,
              "paddingLeft": 1,
            }
          }
          testID="phone_test"
          underlineColorAndroid="transparent"
          value=""
        />
      </View>
      <Text
        style={
          Array [
            Object {
              "marginTop": 15,
            },
            Object {
              "color": "#007c7a",
              "fontSize": 16,
            },
          ]
        }
      >
        District
      </Text>
      <View
        style={
          Object {
            "alignSelf": "center",
            "borderColor": "#007c7a",
            "borderRadius": 10,
            "borderWidth": 1,
            "marginLeft": 20,
            "marginRight": 20,
            "marginTop": 10,
            "width": 300,
          }
        }
      >
        <View
          style={
            Array [
              Object {
                "marginTop": 15,
              },
              Object {
                "alignItems": "center",
                "borderColor": "#007c7a",
                "borderRadius": 20,
                "borderWidth": 1,
                "flexDirection": "row",
                "margin": 10,
                "padding": 5,
              },
            ]
          }
        >
          <RNCPicker
            items={
              Array [
                Object {
                  "label": "Select  District",
                  "testID": undefined,
                  "textColor": 4289374890,
                  "value": "disabled",
                },
                Object {
                  "label": "Ampara",
                  "testID": undefined,
                  "textColor": undefined,
                  "value": "Ampara",
                },
                Object {
                  "label": "Anuradhapura",
                  "testID": undefined,
                  "textColor": undefined,
                  "value": "Anuradhapura",
                },
                Object {
                  "label": "Batticaloa",
                  "testID": undefined,
                  "textColor": undefined,
                  "value": "Batticaloa",
                },
                Object {
                  "label": "Polonnaruwa",
                  "testID": undefined,
                  "textColor": undefined,
                  "value": "Polonnaruwa",
                },
                Object {
                  "label": "Hambantota",
                  "testID": undefined,
                  "textColor": undefined,
                  "value": "Hambantota",
                },
                Object {
                  "label": "Mullaitivu",
                  "testID": undefined,
                  "textColor": undefined,
                  "value": "Mullaitivu",
                },
                Object {
                  "label": "Puttalam",
                  "testID": undefined,
                  "textColor": undefined,
                  "value": "Puttalam",
                },
                Object {
                  "label": "Colombo",
                  "testID": undefined,
                  "textColor": undefined,
                  "value": "Colombo",
                },
                Object {
                  "label": "Galle",
                  "testID": undefined,
                  "textColor": undefined,
                  "value": "Galle",
                },
                Object {
                  "label": "Gampaha",
                  "testID": undefined,
                  "textColor": undefined,
                  "value": "Gampaha",
                },
                Object {
                  "label": "Jaffna",
                  "testID": undefined,
                  "textColor": undefined,
                  "value": "Jaffna",
                },
                Object {
                  "label": "Hambantota",
                  "testID": undefined,
                  "textColor": undefined,
                  "value": "Hambantota",
                },
                Object {
                  "label": "Matara",
                  "testID": undefined,
                  "textColor": undefined,
                  "value": "Matara",
                },
                Object {
                  "label": "Kalutara",
                  "testID": undefined,
                  "textColor": undefined,
                  "value": "Kalutara",
                },
                Object {
                  "label": "Matara",
                  "testID": undefined,
                  "textColor": undefined,
                  "value": "Matara",
                },
                Object {
                  "label": "Kandy",
                  "testID": undefined,
                  "textColor": undefined,
                  "value": "Kandy",
                },
                Object {
                  "label": "Polonnaruwa",
                  "testID": undefined,
                  "textColor": undefined,
                  "value": "Polonnaruwa",
                },
                Object {
                  "label": "Hambantota",
                  "testID": undefined,
                  "textColor": undefined,
                  "value": "Hambantota",
                },
                Object {
                  "label": "Mullaitivu",
                  "testID": undefined,
                  "textColor": undefined,
                  "value": "Mullaitivu",
                },
                Object {
                  "label": "Puttalam",
                  "testID": undefined,
                  "textColor": undefined,
                  "value": "Puttalam",
                },
                Object {
                  "label": "NuwaraEliya",
                  "testID": undefined,
                  "textColor": undefined,
                  "value": "NuwaraEliya",
                },
                Object {
                  "label": "Trincomalee",
                  "testID": undefined,
                  "textColor": undefined,
                  "value": "Trincomalee",
                },
              ]
            }
            onChange={[Function]}
            selectedIndex={0}
            style={
              Array [
                Object {
                  "height": 216,
                },
                undefined,
              ]
            }
            testID="district_test"
          />
        </View>
      </View>
      <Text
        style={
          Array [
            Object {
              "color": "#007c7a",
              "fontSize": 16,
            },
            Object {
              "marginTop": 15,
            },
          ]
        }
      >
        MedicalHistory
      </Text>
      <View
        style={
          Object {
            "alignItems": "center",
            "borderColor": "#007c7a",
            "borderRadius": 20,
            "borderWidth": 1,
            "flexDirection": "row",
            "marginTop": 5,
            "padding": 5,
          }
        }
      >
        <TextInput
          allowFontScaling={true}
          autoCapitalize="none"
          multiline={true}
          numberOfLines={4}
          onChangeText={[Function]}
          placeholder="Enter  MedicalHistory"
          placeholderTextColor="#666666"
          rejectResponderTermination={true}
          style={
            Object {
              "color": "#000",
              "flex": 1,
              "fontSize": 15,
              "paddingLeft": 20,
            }
          }
          underlineColorAndroid="transparent"
          value=""
        />
      </View>
      <View
        style={
          Object {
            "flexDirection": "row",
            "marginTop": 15,
          }
        }
      >
        <Text
          style={
            Object {
              "color": "#007c7a",
              "fontSize": 16,
            }
          }
        >
          RAT results
        </Text>
      </View>
      <View
        style={
          Object {
            "flexDirection": "row",
          }
        }
      >
        <View
          style={
            Array [
              Object {
                "flex": 0.5,
              },
              Object {
                "alignSelf": "center",
                "borderColor": "#007c7a",
                "borderRadius": 10,
                "borderWidth": 1,
                "marginLeft": 5,
                "marginRight": 5,
                "marginTop": 10,
                "width": 300,
              },
            ]
          }
        >
          <View
            style={
              Object {
                "alignItems": "center",
                "borderColor": "#007c7a",
                "borderRadius": 20,
                "borderWidth": 1,
                "flexDirection": "row",
                "margin": 10,
                "padding": 5,
              }
            }
          >
            <RNCPicker
              items={
                Array [
                  Object {
                    "label": "Select",
                    "testID": undefined,
                    "textColor": 4289374890,
                    "value": "disabled",
                  },
                  Object {
                    "label": "POSITIVE",
                    "testID": undefined,
                    "textColor": undefined,
                    "value": "1",
                  },
                  Object {
                    "label": "NAGEATIVE",
                    "testID": undefined,
                    "textColor": undefined,
                    "value": "0",
                  },
                ]
              }
              onChange={[Function]}
              selectedIndex={0}
              style={
                Array [
                  Object {
                    "height": 216,
                  },
                  undefined,
                ]
              }
              testID="result_test"
            />
          </View>
        </View>
      </View>
      <View
        accessible={true}
        focusable={true}
        onClick={[Function]}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          Object {
            "alignItems": "center",
            "backgroundColor": "#009387",
            "borderColor": "#20d1ce",
            "borderRadius": 10,
            "borderWidth": 2,
            "marginBottom": 20,
            "marginLeft": 40,
            "marginRight": 40,
            "marginTop": 60,
            "opacity": 1,
            "paddingHorizontal": 10,
            "paddingVertical": 10,
          }
        }
        testID="admit"
      >
        <Text
          style={
            Object {
              "color": "#fff",
              "fontSize": 20,
            }
          }
        >
          Admit
        </Text>
      </View>
    </View>
  </RCTScrollView>
</RCTSafeAreaView>
`;
